ERCS/erc-7806.md

```diff
@@ -0,0 +1,451 @@
+  is usually a centralized service because it needs to either trust the account owner to repay the sponsored gas.
+  Bundlers are running on MEV based and accounts need to trust bundler providers to include their user operations in the
+  transaction.
+
+[ERC-7521](./eip-7521.md) discusses an SCA solution with intent-centric design. It enables solvers to fulfill account
+owners' intents by acting "on behalf of" account owners. ERC-7521 allows arbitrary intents and verification logics while
+trying to keep the whole solution forward-compatible.
+
+With `SET_CODE_TX_TYPE=0x04` allowing EOA accounts to set contract code, EOA accounts will gain similar programmability
+as SCA (Smart Contact Account). A new standard, that provides EOA with highly-demanded AA features while potentially
+resolving some of the above challenges, will help bring in seamless user experience and drive mass adoptions.
+
+The above EIPs inspired the author to build an intent-centric standard for EOA smart accounts.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT
+RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### UserIntent bytes
+
+Each intent is a packed data structure containing sufficient information about operations the account owner wants to
+execute.
+
+| Field          | Type      | Description                                                                                                                                              |
+|----------------|-----------|----------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `sender`       | `address` | The wallet making the intent                                                                                                                             |
+| `standard`     | `address` | The `IStandard` implementation to validate and parse this `UserIntent`                                                                                   |
+| `header`       | `bytes`   | The metadata of this `UserIntent`, used by `standard`, using `bytes` type to keep flexibility                                                            |
+| `instructions` | `bytes[]` | The detailed content of this `UserIntent`, used by `standard` to determine the `Operation`s need to be executing, using `bytes` type to keep flexibility |
+| `signatures`   | `bytes[]` | Validatable signatures, used by `standard`                                                                                                               |
+
+But using `bytes[]` on functional interface is heavy especially when multiple intents are compound together. Therefore,
+the "UserIntent" object is designed to be a pack-encoded `bytes calldata userIntent` contains the above information
```

Comment:
This paragraph is mixing rationale with specification. In the Specification section, you should only describe the behaviour:

```suggestion
The `UserIntent` object is pack-encoded into `bytes calldata userIntent` containing the above information as shown below:
```

Then the reasoning can be moved into the Rationale section.

Reply:
Updated 👍